wsmaster/codenvy-hosted-platform-api-impl/src/test/java/com/codenvy/api/workspace/LimitsCheckingWorkspaceManagerTest.java

```diff
@@ -454,11 +469,88 @@ public void shouldPermitToCheckRamOnlyForFiveThreadsAtTheSameTime() throws Excep
+                                                                                              null,
+                                                                                              null,
+                                                                                              environmentParser,
+                                                                                              false,
+                                                                                              false,
+                                                                                              2000));
+        doReturn(ImmutableList.of(mock(WorkspaceImpl.class), mock(WorkspaceImpl.class))) // <- currently started 2 workspaces
+                                                                                         .when(manager)
+                                                                                         .getByNamespace(anyString());
+        final WorkspaceCallback callback = mock(WorkspaceCallback.class);
+
+        manager.checkStartedWorkspacesNumberAndPropagateStart("user123", callback);
+
+        verify(callback).call();
+        verify(manager, never()).getByNamespace(any());
+    }
+
+    @Test
+    public void shouldCallCreateCallBackIfEverythingIsOkayWithStartedWorkspacesNumberLimit() throws Exception {
+        final LimitsCheckingWorkspaceManager manager = spy(new LimitsCheckingWorkspaceManager(2,
+                                                                                              2, // max started workspaces limit
+                                                                                              "1gb",
+                                                                                              0,
+                                                                                              null,
+                                                                                              null,
+                                                                                              null,
+                                                                                              null,
+                                                                                              null,
+                                                                                              null,
+                                                                                              environmentParser,
+                                                                                              false,
+                                                                                              false,
+                                                                                              2000));
+        doReturn(emptyList()).when(manager).getByNamespace(anyString()); // <- currently started 0 workspaces
```

Comment:
It means that there are no any workspaces in given namespace (running or stopped). It would better to add test that will test counting all except stopped workspaces